sdformat_mjcf/src/sdformat_mjcf/converters/geometry.py

```diff
@@ -0,0 +1,122 @@
+        geom.type = "box"
+        geom.size = su.vec3d_to_list(box_shape.size() / 2.0)
+    elif sdf_geom.capsule_shape():
+        capsule_shape = sdf_geom.capsule_shape()
+        geom.type = "capsule"
+        geom.size = [capsule_shape.radius(), capsule_shape.length() / 2.0]
+    elif sdf_geom.cylinder_shape():
+        cylinder_shape = sdf_geom.cylinder_shape()
+        geom.type = "cylinder"
+        geom.size = [cylinder_shape.radius(), cylinder_shape.length() / 2.0]
+    elif sdf_geom.ellipsoid_shape():
+        ellipsoid_shape = sdf_geom.ellipsoid_shape()
+        geom.type = "ellipsoid"
+        geom.size = su.vec3d_to_list(ellipsoid_shape.radii())
+    elif sdf_geom.plane_shape():
+        plane_shape = sdf_geom.plane_shape()
+        geom.type = "plane"
+        # The third element of size defines the spacing between square grid
+        # lines for rendering.
+        # TODO (azeey) Consider making this configurable
+        geom.size = su.vec2d_to_list(plane_shape.size() / 2.0) + [0]
+    elif sdf_geom.sphere_shape():
+        sphere_shape = sdf_geom.sphere_shape()
+        geom.type = "sphere"
+        geom.size = [sphere_shape.radius()]
+    elif sdf_geom.mesh_shape():
+        raise RuntimeError("Meshes are not yet supported")
+    else:
+        raise RuntimeError(
+            f"Encountered unsupported shape type {sdf_geom.type()}")
+
+    return geom
+
+
+def add_collision(body, col, pose_resolver=su.pose_resolver):
+    """
+    Converts an SDFormat collision to an MJCF geom and add it to the given
+    body. To differentiate Collision geoms from Visual geoms, we assign
+    Collision geoms group `COLLISION_GEOM_GROUP`.
+
+    :param mjcf.Element body: The MJCF body to which the geom is added.
+    :param sdformat.Collision col: Collision object to be converted.
+    :param pose_resolver: Function to resolve the pose of a
+    sdformat.SemanticPose object.
+    :return: The newly created MJCF geom.
+    :rtype: mjcf.Element
+    """
+    sem_pose = col.semantic_pose()
+    pose = pose_resolver(sem_pose)
+    geom = add_geometry(body, col.name(), pose, col.geometry())
+    geom.group = COLLISION_GEOM_GROUP
+    return geom
+
+
+def add_visual(body, col, pose_resolver=su.pose_resolver):
+    """
+    Converts an SDFormat collision to an MJCF geom and add it to the given
+    body. To differentiate Collision geoms from Visual geoms, we assign
+    Collision geoms group `COLLISION_GEOM_GROUP`.
+
+    :param mjcf.Element body: The MJCF body to which the geom is added.
+    :param sdformat.Collision col: Collision object to be converted.
+    :param pose_resolver: Function to resolve the pose of a
+    sdformat.SemanticPose object.
+    :return: The newly created MJCF geom.
+    :rtype: mjcf.Element
+    """
+    sem_pose = col.semantic_pose()
+    pose = pose_resolver(sem_pose)
+    geom = add_geometry(body, col.name(), pose, col.geometry())
+    geom.group = VISUAL_GEOM_GROUP
+    return geom
```

Comment:
```suggestion
def add_visual(body, vis, pose_resolver=su.pose_resolver):
    """
    Converts an SDFormat visual to an MJCF geom and add it to the given
    body. To differentiate Collision geoms from Visual geoms, we assign
    Visual geoms group `VISUAL_GEOM_GROUP`.

    :param mjcf.Element body: The MJCF body to which the geom is added.
    :param sdformat.Visual vis: Visual object to be converted.
    :param pose_resolver: Function to resolve the pose of a
    sdformat.SemanticPose object.
    :return: The newly created MJCF geom.
    :rtype: mjcf.Element
```
    """
    sem_pose = vis.semantic_pose()
    pose = pose_resolver(sem_pose)
    geom = add_geometry(body, vis.name(), pose, vis.geometry())
    geom.group = VISUAL_GEOM_GROUP
    return geom
```

Reply:
Done in https://github.com/gazebosim/gz-mujoco/pull/2/commits/5af05fddd1f4693e261da624f2a30c37a638db91